### PR TITLE
util/vitess: migrate test-infra to testify

### LINKED ISSUE #26430: migrate test-infra to testify for `util/vitess` pkg

### PR DESCRIPTION
**Pull request by qiaocco**
### What problem does this PR solve?

Issue Number: close #26430 

### Release note 

- No release note

### CODE REVIEW

File: util/vitess/vitess_hash_test.go [hunk 1]
```diff
@@ -20,47 +20,38 @@ import (
...
-	var keybytes [8]byte
-	binary.BigEndian.PutUint64(keybytes[:], value)
-	return strings.ToUpper(hex.EncodeToString(keybytes[:]))
-}
-
-var _ = Suite(&testVitessSuite{})
-
-func (s *testVitessSuite) TestVitessHash(c *C) {
-	defer testleak.AfterTest(c)()
+	defer testleak.AfterTestT(t)()
```
Comment (xhebox) [inline util/vitess/vitess_hash_test.go:30]: Not needed once you have `goleak`.

**Review by tisonkun — COMMENTED** [review]
Second @xhebox , no need for `defer testleak.AfterTestT(t)()`. Rest LGTM.

File: util/vitess/vitess_hash_test.go (hunk 1 above)
Comment (qiaocco) [reply util/vitess/vitess_hash_test.go]: I have removed it.

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
cc @mmyj

**Review by mmyj — APPROVED** [review]
lgtm

**Pull request merged**